test/passing/dune.inc

```diff
@@ -1534,6 +1534,16 @@
  (alias runtest)
  (action (diff label_option_default_args.ml.ref label_option_default_args.ml.output)))
 
+(rule
+ (deps .ocamlformat )
+ (action
+   (with-outputs-to large_string.ml.output
+     (run %{bin:ocamlformat} --profile=janestreet %{dep:large_string.ml}))))
```

Comment:
how long does this test take to run? also I'm wondering if it makes sense to generate the string instead of storing it?

Reply:
It's almost instant for me, and I'm running on a slow vm

Reply:
What do you think of generating this test, either as a dune rule or as a cram test?

Reply:
It looks to be difficult to generate a string that is already formatted (-n 1), maybe that would be enough to have a unit test for list_pn instead of testing the absence of exception of the ml file?

Reply:
ha yes I didn't realize the file was already format. let's scratch that idea and keep the huge file. the unit test is complementary for that, it won't check the resources, just the result. so it all seems good!

Reply:
I added some cram test that passes with this branch but not on master, but if we increase again the length of the string we get a segment fault this time, but this one is probably out of our control and machine dependent

Reply:
can you repro the segfault locally? getting a stacktrace should give us some info about that

Reply:
Yes I can reproduce it with `echo "let _ = \"$(printf '%*s' 300000 | sed 's/ /_ _/g')\"" | ocamlformat - --impl -o /dev/null --debug` but I don't get any trace

Reply:
OK I'll try if this can be fixed with a different encoding.

Reply:
I've got a fix for this, I'll open a PR

Reply:
Done in #1568. Then you can write:

```ocaml
       let l =
         let rec aux (prev, acc) = function
          | [] -> acc
           | [xI] -> aux (xI, (Some prev, xI, None) :: acc) []
           | xI :: (xJ :: _ as xJN) ->
               aux (xI, (Some prev, xI, Some xJ) :: acc) xJN
         in
         aux (x1, [(None, x1, Some x2)]) x2N
       in
      List.rev_map l ~f:(fun (prev, x, next) ->
          lazy_ (fun () -> pp ~prev x ~next) )
      |> sequence
```

Reply:
Thanks, I rebased this one, the CI is now ok!